### DISCUSSION TITLE
Single-layer web app template

### DISCUSSION THREAD

**Discussion by StefanIvovic** [abpframework/abp#12797] [General] [answered]
Hi, i was pleasantly surprised to see single layer template for simple web apps.  I was wondering if there will be tutorial on documentation page for this type of app?

**Comment by EngincanV** [comment, marked as answer]
Yes, actually we are currently preparing documentations for **Application (Single-Layer) Startup Template** (`app-nolayers`) and they are under review (and will be available asap). You can check the following PRs, if you want to read/examine them now:

- https://github.com/abpframework/abp/pull/12610
- https://github.com/abpframework/abp/pull/12640
- https://github.com/abpframework/abp-samples/pull/166

**Reply by StefanIvovic** [reply]
Cool, thanks for info rly appreciate apb teams work

**Comment by StefanIvovic** [comment]
Is it possible to add DbMigrator project to single layer app ?

**Reply by EngincanV** [reply]
We don't plan to add it to **Application (Single Layer) Template** but if you want you can add it for your own project by examining the **Application Template**. But I don't recommend it. Instead of adding a new project just for creating the database and seeding initial data, you can use the `dotnet run --migrate-database` command.

**Reply by StefanIvovic** [reply]
yes i agree, but i was using it on multilayer app and i really liked it. And also its much easier to do migration on production with it. Right now when i need to migrate prod db (heroku postgres) i set command line argument to --migrate-database and run app with prod settings in visual studio and it does the job bit it feels so janky. Do you have any suggestions on how to migrate prod db.

**Reply by EngincanV** [reply]
I see your point, in that case, it might be better to create a new DbMigrator project and stick with it by migrating production db.

**Reply by StefanIvovic** [reply]
I did try and it didnt go too well :) Layered app is soo much different then single layered. I was trying to make it work for about 2hrs.

**Reply by StefanIvovic** [reply]
ok so i tried and i get exception in DbMigratorHostedService.StartAsync()                  application.Initialize();
The requested service 'Microsoft.AspNetCore.Hosting.IWebHostEnvironment' has not been registered.

using .net 6
